component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,378 @@
+                        type = {DataType.INT, DataType.LONG, DataType.TIME},
+                        optional = true,
+                        defaultValue = "300000")
+        },
+        examples = {
+                @Example(
+                        syntax = "define window cseEventWindow (symbol string, price float, volume int) " +
+                                "evn:resourceBatch('X', symbol) output all events;\n\n" +
+                                "@info(name = 'query0')\n" +
+                                "from cseEventStream\n" +
+                                "insert into cseEventWindow;\n\n" +
+                                "@info(name = 'query1')\n" +
+                                "from cseEventWindow\n" +
+                                "select symbol, sum(price) as price\n" +
+                                "insert all events into outputStream ;",
+                        description = "This will processing events as batches based on the registered 'X' resource " +
+                                "count as a batch length and 'symbol' as grouping key and out put all events as " +
+                                "chunk once the batches expired."
+                ),
+                @Example(
+                        syntax = "define stream SweetProductDefectsDetector(productId string, colorCode string, " +
+                                "height long, width long);\n" +
+                                "define stream SweetProductDefectAlert(productId string, isDefected bool);\n" +
+                                "\n" +
+                                "@info(name='product_color_code_rule') \n" +
+                                "from SweetProductDefectsDetector#env:resourceIdentifier(\"rule-group-1\")\n" +
+                                "select productId, if(colorCode == '#FF0000', true, false) as isValid\n" +
+                                "insert into DefectDetectionResult;\n" +
+                                "\n" +
+                                "@info(name='product_dimensions_rule') \n" +
+                                "from SweetProductDefectsDetector#env:resourceIdentifier(\"rule-group-1\")\n" +
+                                "select productId, if(height == 5 && width ==10, true, false) as isValid\n" +
+                                "insert into DefectDetectionResult;\n" +
+                                "\n" +
+                                "@info(name='defect_analyzer') \n" +
+                                "from DefectDetectionResult#window.env:resourceBatch(\"rule-group-1\", productId, " +
+                                "60000)\n" +
+                                "select productId, and(not isValid) as isDefected\n" +
+                                "insert into SweetProductDefectAlert;",
+                        description = "This example demonstrate the usage of 'ResourceBatchWindowProcessor' " +
```

Comment:
No one knows that's "ResourceBatchWindowProcessor" in the docs its just  "resourceBatch" window. rephrase the text same in other places don't use implementation names

Reply:
Fixes added: d49eb00ce0888f34a4d58f8383894d3ea2f52bcf